monitoring.go

```diff
@@ -214,17 +215,40 @@ func (sc *snowflakeConn) getQueryResultResp(
 		headers[headerAuthorizationKey] = fmt.Sprintf(headerSnowflakeToken, token)
 	}
 	url := sc.rest.getFullURL(resultPath, &param)
-	res, err := sc.rest.FuncGet(ctx, sc.rest, url, headers, sc.rest.RequestTimeout)
-	if err != nil {
-		logger.WithContext(ctx).Errorf("failed to get response. err: %v", err)
-		return nil, err
-	}
-	defer res.Body.Close()
+
 	var respd *execResponse
-	if err = json.NewDecoder(res.Body).Decode(&respd); err != nil {
-		logger.WithContext(ctx).Errorf("failed to decode JSON. err: %v", err)
-		return nil, err
+	retry := 0
+	retryPattern := []int32{1, 1, 2, 3, 4, 8, 10}
```

Comment:
As long as this logic is copied, maybe we can refactor this code to common function instead of copying?

Reply:
Refactored to `func getQueryResult`

Reply:
What do you think about `getQueryResultWithRetriesForAsyncMode`? `getQueryResult` seems to little specific.

Reply:
Renamed